src/components/Launch/LaunchWorkflowForm/inputHelpers/blob.ts

```diff
@@ -0,0 +1,88 @@
+import { Core } from 'flyteidl';
+import { isObject } from 'lodash';
+import { BlobDimensionality } from 'models';
+import { BlobValue, InputValue } from '../types';
+import { literalNone } from './constants';
+import { ConverterInput, InputHelper, InputValidatorParams } from './types';
+
+function fromLiteral(literal: Core.ILiteral): InputValue {
+    if (!literal.scalar || !literal.scalar.blob) {
+        throw new Error('Literal blob missing scalar.blob property');
+    }
+    const { blob } = literal.scalar;
+    const uri = blob.uri ?? '';
+    // throw away empty format values
+    const format = blob.metadata?.type?.format || undefined;
+    const dimensionality =
+        blob.metadata?.type?.dimensionality ?? BlobDimensionality.SINGLE;
+
+    return {
+        dimensionality,
+        format,
+        uri
+    };
+}
+
+// Allows for string values ('single'/'multipart') when specifying blobs manually in collections
+function getDimensionality(value: string | number) {
+    return typeof value === 'number'
+        ? value
+        : BlobDimensionality[
+              value.toUpperCase() as keyof typeof BlobDimensionality
+          ];
+}
```

Comment:
Just a thought, we may be able to utilize more type guards here:

```tsx
export function isKeyOfBlobDimensionality(val: string): val is keyof typeof BlobDimensionality {
    return Object.keys(BlobDimensionality).indexOf(val) >= 0;
}

// Allows for string values ('single'/'multipart') when specifying blobs manually in collections
function getDimensionality(value: string | number) {
    if (typeof value === 'number') {
        return value;
    }
    if (isKeyOfBlobDimensionality(value)) {
        return BlobDimensionality[value];
    }
    // Note unclear what we would want to do in this situation
    return SOME_DEFAULT;
}
```

Reply:
Typeguard is a good idea. I'm not sure what to do with a default either. The code which calls this function throws an exception if an invalid value is returned, so I don't think this function is expected to coerce an invalid value into a sensible default. Doing so might cause issues which are hard to diagnose by treating a URI specifying a directory as a single file and only processing the first file in the directory
So the preference is to completely fail if a bad value is provided.

Reply:
It turns out there are two places which call this code. One does assume the value has already been validated. But either way, the result of indexing into an enum with a string that is _not_ a key is that it will return the first item in the enum. (Thanks, Typescript?) So I can preserve that behavior.